torch/fx/subgraph_rewriter.py

```diff
@@ -83,7 +85,114 @@ def attributes_are_equal(pn : Node, gn : Node) -> bool:
         return True
 
 
-def replace_pattern(gm : GraphModule, pattern : Callable, replacement : Callable) -> List[Match]:
+def _replace_submodules(gm: GraphModule, replacement: torch.nn.Module) -> None:
+    gm.delete_all_unused_submodules()
+
+    class ErrType(Enum):
+        EXISTS_AS_OTHER_ATTR = auto()
+        DOES_NOT_EXIST = auto()
+
+    class FoundModule():
+        def __init__(self, mod_match: Optional[torch.nn.Module],
+                     err: Optional[ErrType]):
+            assert(mod_match is not None or err is not None)
+            self.mod_match = mod_match
+            self.err = err
+
+        # For tuple unpacking
+        def __iter__(self):
+            yield from [self.mod_match, self.err]
+
+    def try_get_submodule(mod: torch.nn.Module, target: str) -> FoundModule:
+        try:
+            mod_match = mod.get_submodule(target)
+            return FoundModule(mod_match, None)
+        except AttributeError as e:
+            assert(e.args and len(e.args))
+            if ("not an nn.Module" in e.args[0]):
+                return FoundModule(None, ErrType.EXISTS_AS_OTHER_ATTR)
+            elif("has no attribute" in e.args[0]):
+                return FoundModule(None, ErrType.DOES_NOT_EXIST)
+            else:
+                raise RuntimeError("Unknown error type found when "
+                                   "attempting to replace missing "
+                                   "submodule in subgraph rewriter!"
+                                   "Please report a bug to PyTorch")
```

Comment:
probably want to include `str(e)` in the message so we can diagnose bug reports more easily